generic_expression_patterns_modules/process.py

```diff
@@ -983,9 +956,418 @@ def concat_simulated_data_columns(local_dir, num_runs, project_id, data_type):
+
+    fig_abs = sns.scatterplot(
+        data=merged_df,
+        x=f"Test statistic (Real)_grp_{condition_1}",
+        y=f"Test statistic (Real)_grp_{condition_2}",
+        hue="max Z score",
+        size="max Z score",
+        linewidth=0,
+        alpha=0.7,
+        ax=axes[0],
+        palette=cmap,
+    )
+    fig_abs.plot([0, 4], [0, 4], "--k")
+
+    fig_raw = sns.scatterplot(
+        data=merged_df,
+        x=f"Test statistic (Real)_grp_{condition_1}_raw",
+        y=f"Test statistic (Real)_grp_{condition_2}_raw",
+        hue="max Z score",
+        size="max Z score",
+        linewidth=0,
+        alpha=0.7,
+        ax=axes[1],
+        palette=cmap,
+    )
+    fig_raw.plot([-4, 4], [-4, 4], "--k")
+
+    # Add labels
+    fig.suptitle(f"({xlabel}) vs ({ylabel})", fontsize=16)
+    fig.text(0.5, 0.04, xlabel, ha="center", va="center")
+    fig.text(0.06, 0.5, ylabel, ha="center", va="center", rotation="vertical")
+    axes[0].set_title("using abs(log2FC)")
+    axes[1].set_title("using log2FC")
+    axes[0].set_xlabel("")
+    axes[1].set_xlabel("")
+    axes[0].set_ylabel("")
+    axes[1].set_ylabel("")
+    print(fig)
+
+    # ADD NEWLINE TO TITLE
+    # MAKE LABELS LARGER
+    # MOVE LEGEND?
+
+
+def get_and_save_DEG_lists(merged_one_condition_df, condition):
+    """
+    Get list of DEGs using traditional criteria (log2FC and p-value)
+    and using z-score cutoff. Return different combinations of gene
+    lists.
+
+    Arguments
+    ---------
+    merged_one_condition_df: df
+        df of results for one of the E-GEOD-33245 conditions ('1v2', '1v3', '1v4' or '1v5')
+        returned from `merge_abs_raw_dfs`
+    condition: str
+        Condition from E-GEOD-33245 associated with 'merged_one_condition_df'.
+        Either '1v2', '1v3', '1v4' or '1v5'
+    """
+    # Get DEGs using traditional criteria
+    degs_traditional = list(
+        (
+            merged_one_condition_df[
+                (merged_one_condition_df[f"Test statistic (Real)_grp_{condition}"] > 1)
+                & (
+                    merged_one_condition_df[f"Adj P-value (Real)_grp_{condition}"]
+                    < 0.05
+                )
+            ]
+            .set_index("Gene ID")
+            .index
+        )
+    )
+    print(f"No. of DEGs using traditional criteria: {len(degs_traditional)}")
+
+    # Get predicted specific DEGs using z-score cutoff
+    degs_specific = list(
+        (
+            merged_one_condition_df[
+                (merged_one_condition_df[f"Test statistic (Real)_grp_{condition}"] > 1)
+                & (merged_one_condition_df[f"Z score_grp_{condition}"].abs() > 4.44)
+            ]
+            .set_index("Gene ID")
+            .index
+        )
+    )
+    print(f"No. of specific DEGs using z-score: {len(degs_specific)}")
+
+    # Get predicted generic DEGs using z-score cutoff
+    # Z-score cutoff was found by calculating the score
+    # whose invnorm(0.05/5549)
```

Comment:
I assume 5549 is the number of tests you're doing, is that right? So this is essentially a cutoff of p = 0.05, with a Bonferroni correction for 5549 tests?

Reply:
yes. Here we are trying to get an equivalent z-value cutoff to match the adjusted p-value cutoff. I'll add a note here to clarify